static_src/stores/base_store.js

```diff
@@ -9,6 +9,10 @@ import Immutable from 'immutable';
 import AppDispatcher from '../dispatcher.js';
 import LoadingStatus from '../util/loading_status.js';
 
+// TODO clean up the listeners and reduce the MAX_LISTENERS to 15
+const MAX_LISTENERS = 20;
+const MAX_LISTENERS_THRESHOLD = 10;
```

Comment:
You mean why not set an unlimited number of listeners? In this case, we have a dynamic number of ActivityLogItems on the page, so it seems strange to me that the number of listeners explodes on the app page. That's the scenario I'm trying to catch, but you're right that it's not a leak so why should we care?

It feels anti-flux to me, since the container component's are responsible to listening to state changes and passing that down the component hierarchy. I think it's reasonable that each panel acts as a component and would each could have it's own listeners but I don't think smaller components should listen to the stores.